test/functional/legacy/053_string_and_html_objects_et_alia_spec.lua

```diff
@@ -0,0 +1,216 @@
+-- Tests for string and html text objects.
+-- Note that the end-of-line moves the cursor to the next test line.
+-- Also test match() and matchstr()
+-- Also test the gn command and repeating it.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+local eq, eval = helpers.eq, helpers.eval
+
+local function expect_line(text)
+  return eq(text, eval('getline(".")'))
+end
+
+local function expect_line_above(text)
+  return eq(text, eval('getline(line(".")-1)'))
+end
+
+describe('text objects:', function()
+  before_each(clear)
+
+  it('quote objects', function()
+    insert([[
+      start: "wo\"rd\\" foo
+      'foo' 'bar' 'piep'
+      bla bla `quote` blah
+      out " in "noXno"
+      "'" 'blah' rep 'buh'
+      bla `s*`d-`+++`l**` b`la
+      voo "nah" sdf " asdf" sdf " sdf" sd]])
+    execute('/^start:/')
+    feed('da"<cr>')
+    feed("0va'a'rx<cr>")
+    feed('02f`da`<cr>')
+    feed('0fXdi"<cr>')
+    feed("03f'vi'ry<cr>")
+    execute('set quoteescape=+*-')
+    feed('di`<cr>')
+    feed('$F"va"oha"i"rz<cr>')
+    expect([[
+      start: foo
+      xxxxxxxxxxxx'piep'
+      bla bla blah
+      out " in ""
+      "'" 'blah'yyyyy'buh'
+      bla `` b`la
+      voo "zzzzzzzzzzzzzzzzzzzzzzzzzzzzsd]])
+  end)
+
+  it('html objects', function()
+    insert([[
+      <begin>
+      -<b>asdf<i>Xasdf</i>asdf</b>-
+      -<b>asdX<i>a<i />sdf</i>asdf</b>-
+      -<b>asdf<i>Xasdf</i>asdf</b>-
+      -<b>asdX<i>as<b />df</i>asdf</b>-
+      -<b>
+      innertext object
+      </b>
+      </begin>]])
+    execute('/^<begin')
+    feed('jfXdit<cr>')
+    feed('0fXdit<cr>')
+    feed('fXdat<cr>')
+    feed('0fXdat<cr>')
+    expect([[
+      <begin>
+      -<b>asdf<i></i>asdf</b>-
+      -<b></b>-
+      -<b>asdfasdf</b>-
+      --
+      -<b>
+      innertext object
+      </b>
+      </begin>]])
+    feed('dit<cr>') -- TODO
```

Comment:
This is a bug in the original test. `<cr>` moves to the first non-blank, there's no way for `dit` to work correctly. We can just fix the test in this case (e.g. use `j` instead of `<cr>`).